wasp/widgets.py

```diff
@@ -244,6 +250,18 @@ def draw(self):
         x = self._x
         y = self._y
         color = self._color
+        if self._color is None:
+            self._color = wasp.system.theme('slider-default')
+            color = self._color
+            # Automatically generate a lowlight color
+            if color < 0b10110_000000_00000:
```

Comment:
I don't think you need to pull this out into a function. I think you can just `if self._lowlight is None:' here and set it to  None from `self.__init__()`

Reply:
no need to check that - if self._color is none, lowlight has to be none too.

Reply:
What about the other case then self._color is not None?

We can either calculate it in two places: `__init__` and `draw` or we can calculate it only in `draw`. If we calculate it only in `draw` then there is no need to pull it out into a function.

Reply:
we can set it so that it only generates it once in `draw`

Reply:
please review the current implementation